### ISSUE TITLE
Gatsby + storybook + async/await + NODE_ENV=production == FAIL

### ISSUE THREAD

**Issue by i8ramin** [gatsbyjs/gatsby#13176]
## Description

For some reason, the following combination of things causes the Gatsby build to fail:

- `async/await` code
- `@storybook` packages
- `NODE_ENV=production`

### Steps to reproduce

Repo with basic Gatsby setup + @storybook devDependencies + async/await code:
https://github.com/i8ramin/gatsby-test-prod-async

```bash
git clone https://github.com/i8ramin/gatsby-test-prod-async
cd gatsby-test-prod-async
NODE_ENV=production yarn install --ignore-optional
NODE_ENV=production yarn build
```

### Possibly related GH issues

- https://github.com/storybooks/storybook/issues/5510
- https://github.com/storybooks/storybook/issues/537
- https://github.com/storybooks/storybook/issues/489

### Expected result

Expect the site to build just fine. Which it does if you do any of the following:

- remove the `async/await` code, it builds just fine.
- remove the **@storybook** from `devDependencies`, it builds just fine.
- add **@storybook** modules to `dependencies`, it builds just fine (after issuing `rm -rf node_modules && NODE_ENV=production yarn install --ignore-optional`)

### Actual result

```
error Generating JavaScript bundles failed


  Error: ./src/pages/index.js
  Module not found: Error: Can't resolve 'regenerator-runtime/runtime' in '/Users/home/tmp/gatsby-test-prod-  async/src/pages'
  resolve 'regenerator-runtime/runtime' in '/Users/home/tmp/gatsby-test-prod-async/src/pages'
    Parsed request is a module
    using description file: /Users/home/tmp/gatsby-test-prod-async/package.json (relative path: ./src/pages)      Field 'browser' doesn't contain a valid alias configuration
      resolve as module
        /Users/home/tmp/gatsby-test-prod-async/src/pages/
```

### Environment

  System:
    OS: macOS 10.14.3
    CPU: (8) x64 Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz
    Shell: 5.7.1 - /usr/local/bin/zsh
  Binaries:
    Node: 11.6.0 - ~/.nvm/versions/node/v11.6.0/bin/node
    Yarn: 1.12.3 - /usr/local/bin/yarn
    npm: 6.5.0-next.0 - ~/.nvm/versions/node/v11.6.0/bin/npm
  Languages:
    Python: 2.7.15 - /usr/local/bin/python
  Browsers:
    Chrome: 73.0.3683.103
    Firefox: 54.0.1
    Safari: 12.0.3
  npmPackages:
    gatsby: ^2.3.5 => 2.3.14 
    gatsby-plugin-root-import: 2.0.5 => 2.0.5 

> Related issue created in the @storybook repo: https://github.com/storybooks/storybook/issues/6443

**Comment by i8ramin** [comment]
Btw, adding **regenerator-runtime** to `dependencies` also solves the problem. But its still not clear to me why this is happening in the first place.

Also, does Gatsby need or care about `NODE_ENV=production` in any way? If `gatsby build` always does a **production** build, I guess there is really no reason to set `NODE_ENV=production`, right?

**Comment by trevorlitsey** [comment]
Hi @i8ramin, thanks for the sample repo. On my local machine, I am unfortunately not able to reproduce the error. The build works fine for me.

That said, I'm not sure that one should necessarily expect a build to be successful if only production dependencies are installed. `NODE_ENV=production yarn install`, as you may know, only installs packages listed in `dependencies`, and ignores packages listed in `devDependencies`. I'm not an expert on the gatsby build scripts, but there are a handful of dependencies listed as `devDependencies`. Those libraries are likely needed for the build process, but might not contain any code meant to be shipped to the browser.

https://docs.npmjs.com/cli/install

**Comment by trevorlitsey** [comment]
Hi @i8ramin, sorry, I just realized I provided a link to the NPM docs while your reproduction steps use Yarn. I just cloned the repo again and still no issues with the build when using Yarn.

Same treatment in regards to `devDependencies`, but here's the relevant Yarn docs on `yarn install`:
https://yarnpkg.com/en/docs/cli/install#toc-yarn-install-production-true-false

Closing this issue as I believe the Gatsby code is working as expected.

Best!
